paddle/operators/recurrent_op.h

```diff
@@ -33,7 +34,12 @@ class RecurrentAlgorithm {
   void Run(const framework::Scope& scope,
            const platform::DeviceContext& dev_ctx) const;
 
-  void Init(std::unique_ptr<rnn::Argument> arg) { arg_ = std::move(arg); }
+  void Init(std::unique_ptr<rnn::Argument> arg,
```

Comment:
the `rnn::Argument` is a argument that should be passed to `RecurrentAlgorithm`, a unique pointer may be lighter to pass, and `std::move` is used here, finally only `RecurrentAlgorithm` owns this memory.